README.md

```diff
@@ -737,3 +739,103 @@ pipe(
   Option.option.traverse(TaskEither.taskEither) // this is the same as above.
 )
 ```
+
+## <a name="reader"></a>Reader
+
+If you already had to carry dependencies throught out multiple functions, you will quickly understand the value of `Reader`.
+You can see the definition given by Giulio Canti [here](https://dev.to/gcanti/getting-started-with-fp-ts-reader-1ie5), and basically `Reader<R,A>` represents a function `(r: R) => A` where `R` will be your dependencies, and A the result.
+
+Let's take an example:
+
+```typescript
+const foo = ({
+  firstId,
+  secondId,
+  eventData,
+  firstDependency,
+  secondDependency,
+}): TaskEither<Error, Value> =>
+  pipe(
+    firstDependency.getById(firstId),
+    TaskEither.chain(() => secondDependency.sendEvent({ secondId, eventData }))
+  );
+```
+
+Using `Reader`, you could write it this way:
+
+```typescript
+const foo = ({
+  firstId,
+  secondId,
+  eventData,
+}): ReaderTaskEither<Deps, Error, Value> =>
+  pipe(
+    getById(firstId),
+    ReaderTaskEither.chain(() => sendEvent({ secondId, eventData }))
+  );
+
+// with:
+const getById = (firstId) =>
+  pipe(
+    Reader.ask<Deps>(),
```

Comment:
what is `Deps` here?
I remember asking myself this question when I started learning about Reader

Reply:
Oh right, I forgot to define what it is in this case, I'll add it now!

Reply:
I've updated my PR to try to explicit what was `Deps`, is it better now?